src/Microsoft.ML.Data/Transforms/ValueMappingTransformer.cs

```diff
@@ -0,0 +1,995 @@
+                verReadableCur: 0x00010001,
+                verWeCanReadBack: 0x00010001,
+                loaderSignature: LoaderSignature,
+                loaderAssemblyName: typeof(ValueMappingTransformer).Assembly.FullName);
+        }
+
+        private static VersionInfo GetTermLookupVersionInfo()
+        {
+            return new VersionInfo(
+                modelSignature: "TXTLOOKT",
+                // verWrittenCur: 0x00010001, // Initial.
+                verWrittenCur: 0x00010002, // Dropped sizeof(Float).
+                verReadableCur: 0x00010002,
+                verWeCanReadBack: 0x00010002,
+                loaderSignature: LoaderSignature,
+                loaderAssemblyName: typeof(ValueMappingTransformer).Assembly.FullName);
+        }
+
+        public sealed class Column : OneToOneColumn
+        {
+            public static Column Parse(string str)
+            {
+                var res = new Column();
+                if (res.TryParse(str))
+                    return res;
+                return null;
+            }
+
+            public bool TryUnparse(StringBuilder sb)
+            {
+                Contracts.AssertValue(sb);
+                return TryUnparseCore(sb);
+            }
+        }
+
+        public sealed class Arguments
+        {
+            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "New column definition(s) (optional form: name:src)", ShortName = "col", SortOrder = 1)]
+            public Column[] Column;
+
+            [Argument(ArgumentType.AtMostOnce, HelpText = "The name of the column containing the keys", ShortName = "key")]
+            public string KeyColumn;
+
+            [Argument(ArgumentType.AtMostOnce, HelpText = "The name of the column containing the values", ShortName = "value")]
+            public string ValueColumn;
+
+            [Argument(ArgumentType.Multiple, HelpText = "The data loader", NullName = "<Auto>", SignatureType = typeof(SignatureDataLoader))]
+            public IComponentFactory<IMultiStreamSource, IDataLoader> Loader;
+
+            [Argument(ArgumentType.AtMostOnce, IsInputFileName = true, HelpText = "The data file containing the terms", ShortName = "data", SortOrder = 2)]
+            public string DataFile;
+
+            [Argument(ArgumentType.AtMostOnce,
+                HelpText = "Specifies whether the values are key values or numeric, only valid when loader is not specified and the type of data is not an idv.")]
+            public bool ValuesAsKeyType = true;
+        }
+
+        protected ValueMappingTransformer(IHostEnvironment env, IDataView lookupMap,
+            string keyColumn, string valueColumn, (string Input, string Output)[] columns)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(ValueMappingTransformer)), columns)
+        {
+            env.CheckNonEmpty(keyColumn, nameof(keyColumn), "A key column must be specified when passing in an IDataView for the value mapping");
```

Comment:
>env [](start = 12, length = 3)

you have Host here.

Reply:
What is wrong with having host? Should the variable name be different?

---
In reply to: [243464384](https://github.com/dotnet/machinelearning/pull/1710#discussion_r243464384) [](ancestors = 243464384)

Reply:
my bad, not a great phrasing. Use Host.Check... instead of env.Check.
Env is just some environment passed to you, host is bind to this exact transform, so from reporting view it's better to use Host.

---
In reply to: [243493174](https://github.com/dotnet/machinelearning/pull/1710#discussion_r243493174) [](ancestors = 243493174,243464384)

Reply:
ok thanks for the clairification

---
In reply to: [243502032](https://github.com/dotnet/machinelearning/pull/1710#discussion_r243502032) [](ancestors = 243502032,243493174,243464384)